src/DictList.elm

```diff
@@ -938,6 +965,174 @@ fromDict dict =
 
 
 
+-- LIST EXTRA
+
+
+{-| Extract the last element of a list.
+    last (fromList [(1, 1), (2, 2), (3, 3)]) == Just (3, 3)
+    last (fromList []) == Nothing
+-}
+last : DictList comparable v -> Maybe ( comparable, v )
+last xs =
+  toList xs
+  |> List.Extra.last
+
```

Comment:
This will work, but doing a complete `toList` on the `DictList` will be somewhat inefficient -- we'll look up every key in the `Dict`, when we only really need to look up the last one. So, I'd suggest taking the opportunity to work on the internals of `DictList` directly -- something roughly like:

```elm
last (DictList dict list) =
    List.Extra.last list
        |> Maybe.map (\key -> DictList key (unsafeGet key dict))
```

That is, we can get the last key from the internal list directly, and then use `unsafeGet` to look up just that value (and it's not actually unsafe in this context, as long as our list and dict have remained consistent, which they will have unless we have bugs).